src/decode/lzma.rs

```diff
@@ -559,3 +580,54 @@ impl DecoderState {
         Ok(result)
     }
 }
+
+#[derive(Debug)]
+/// Raw decoder for LZMA.
+pub struct LzmaDecoder {
+    params: LzmaParams,
+    memlimit: usize,
+    decoder: DecoderState,
+}
+
+impl LzmaDecoder {
+    /// Creates a new object ready for decompressing data that it's given for the input
+    /// dict size, expected unpacked data size, and memory limit for the internal buffer.
+    pub fn new(params: LzmaParams, memlimit: Option<usize>) -> error::Result<LzmaDecoder> {
+        Ok(Self {
+            params,
+            memlimit: memlimit.unwrap_or(usize::MAX),
+            decoder: DecoderState::new(params.properties, params.unpacked_size),
+        })
+    }
+
+    /// Performs the equivalent of replacing this decompression state with a freshly allocated copy.
+    ///
+    /// Because the decoder state is reset, the unpacked size may optionally be re-specified. If None
+    /// is given, the previous unpacked size that the decoder was initialized with remains unchanged.
+    ///
+    /// This function may not allocate memory, though, and attempts to reuse any previously existing resources.
+    #[cfg(feature = "raw_decoder")]
+    pub fn reset(&mut self, unpacked_size: Option<Option<u64>>) {
+        self.decoder.reset_state(self.params.properties);
+
+        if let Some(unpacked_size) = unpacked_size {
+            self.decoder.set_unpacked_size(unpacked_size);
+        }
+    }
+
+    /// Decompresses the input data into the output, consuming only as much input as needed and writing as much output as possible.
+    pub fn decompress<'a, W: io::Write, R: io::BufRead>(
+        &mut self,
+        input: &mut R,
+        output: &'a mut W,
```

Comment:
Is this explicit `'a` lifetime needed here? I don't see it anywhere else in this function so I guess it can be removed as per the [lifetime elision rules](https://doc.rust-lang.org/reference/lifetime-elision.html)?